src/chains/algorand_1/templates/chainActions/chainSpecific/asset_destroy.ts

```diff
@@ -0,0 +1,44 @@
+import * as algosdk from 'algosdk'
+import {
+  AlgorandDecomposeReturn,
+  AlgorandChainActionType,
+  AlgorandSuggestedParams,
+  AlgorandTransactionTypeCode,
+  AlgorandTxAction,
+  AlgorandTxActionRaw,
+  AlgorandActionAssetDestroyParams,
+} from '../../../models'
+import { AlgorandActionHelper } from '../../../algoAction'
+import { isNullOrEmpty } from '../../../../../helpers'
+
+/**
+ * Composes asset destroy action */
+export const composeAction = (args: AlgorandActionAssetDestroyParams, suggestedParams: AlgorandSuggestedParams) => {
+  const argsEncodedForSdk = new AlgorandActionHelper(args as AlgorandTxAction).actionEncodedForSdk
+  const { from, note, assetIndex } = argsEncodedForSdk
+  const composedAction = algosdk.makeAssetDestroyTxnWithSuggestedParams(from, note, assetIndex, suggestedParams)
+  return { ...composedAction }
+}
+
+export const decomposeAction = (action: AlgorandTxAction | AlgorandTxActionRaw): AlgorandDecomposeReturn => {
+  const actionHelper = new AlgorandActionHelper(action)
+  const actionParams = actionHelper.paramsOnly
+  // Identify chainActionType using type
+  if (
+    actionParams?.type === AlgorandTransactionTypeCode.AssetConfig &&
+    isNullOrEmpty(actionParams?.assetManager) &&
```

Comment:
Does AssetDestroy really require all these fields? If you are trying to decide which action it is, you only need to look for one field that makes it uniquely AssetDestroy. Or if it could be one or more fields, then you should probably be using an OR list here not an AND

Reply:
None of them needed for destroying asset. Only id. So I'm checking to confirm that all of them are null or empty so that It can only be assetDestroy